pkg/util/kernel/fs.go

```diff
@@ -36,138 +33,3 @@ func ParseMountInfoFile(pid int32) ([]*mountinfo.Info, error) {
-	if err != nil {
-		return nil, err
-	}
-
-	for {
-		for _, m := range mi {
-			if path == m.Mountpoint {
-				return m, nil
-			}
-		}
-
-		if path == "/" {
-			break
-		}
-		path = filepath.Dir(path)
-	}
-
-	return nil, fmt.Errorf("no matching mountpoint found")
-}
-
-// isDebugFSMounted would test the existence of file /sys/kernel/debug/tracing/kprobe_events to determine if debugfs (or debugfs AND tracefs) is/are mounted or not
-// returns a boolean and a possible error message
-func isDebugFSMounted() (bool, error) {
-	_, err := os.Stat("/sys/kernel/debug/tracing/kprobe_events")
-	if err != nil {
-		if os.IsPermission(err) {
-			return false, fmt.Errorf("eBPF not supported, does not have permission to access debugfs")
-		} else if os.IsNotExist(err) {
-			return false, fmt.Errorf("debugfs is not mounted and is needed for eBPF-based checks, run \"sudo mount -t debugfs none /sys/kernel/debug\" to mount debugfs")
-		} else {
-			return false, fmt.Errorf("an error occurred while accessing debugfs: %w", err)
-		}
-	}
-
-	mi, err := GetMountPoint("/sys/kernel/debug/tracing/kprobe_events")
-	if err != nil {
-		return false, fmt.Errorf("unable to detect debugfs mount point: %w", err)
-	}
-
-	if mi.FSType != "tracefs" && mi.FSType != "debugfs" {
-		return false, fmt.Errorf("kprobe_events mount point(%s): wrong fs type(%s)", mi.Mountpoint, mi.FSType)
-	}
-
-	// on fargate, kprobe_events is mounted as ro
-	if !fargate.IsFargateInstance() {
-		options := strings.Split(mi.Options, ",")
-		for _, option := range options {
-			if option == "ro" {
-				return false, fmt.Errorf("kprobe_events mount point(%s) is in read-only mode", mi.Mountpoint)
-			}
-		}
-	}
-
-	return true, nil
-}
-
-// isTraceFSMounted would test the existence of file /sys/kernel/tracing/kprobe_events to determine if tracefs is mounted (and debugfs is not)
-// returns a boolean and a possible error message
-func isTraceFSMounted() bool {
-	_, err := os.Stat("/sys/kernel/tracing/kprobe_events")
-	if err != nil {
-		if os.IsPermission(err) {
-			seclog.Infof("eBPF not supported, does not have permission to access tracefs kprobe_events (/sys/kernel/tracing/kprobe_events)")
-			return false
-		} else if os.IsNotExist(err) {
-			seclog.Infof("tracefs is not mounted in /sys/kernel/tracing/ and is needed for eBPF-based checks, run \"sudo mount -t tracefs none /sys/kernel/tracing\" to mount tracefs")
-			return false
-		} else {
-			seclog.Infof("an error occurred while accessing tracefs sysfs entry (/sys/kernel/tracing/kprobe_events): %s", err)
-			return false
-		}
-	}
-
-	mi, err := GetMountPoint("/sys/kernel/tracing/kprobe_events")
-	if err != nil {
-		seclog.Infof("unable to detect tracefs mount point (/sys/kernel/tracing/): %s", err)
-		return false
-	}
-
-	if mi.FSType != "tracefs" {
-		seclog.Infof("kprobe_events mount point(%s): wrong fs type(%s)", mi.Mountpoint, mi.FSType)
-		return false
-	}
-
-	// on fargate, kprobe_events is mounted as ro
-	if !fargate.IsFargateInstance() {
```

Comment:
Was this part related to Fargate moved to ebpf-manager ?

Reply:
@lebauce Let me know if you want changes to be made, or if I can merge.

Reply:
You're right, the error when writing is probably enough. Let's merge it this way